backend/src/main/java/com/empire/World.java

```diff
@@ -3120,6 +3113,11 @@ public boolean isHarvestTurn() {
 	public int turnsUntilHarvest() {
 		return (13 - ((date + 52 - 12) % 13)) % 13;
 	}
+
+	@Override
+	public double getGoodwill(String nation) {
```

Comment:
This method makes it look to me like the NationData class should be the one implementing GoodwillProvider